doc/specs/wiki.md

```diff
@@ -0,0 +1,391 @@
+# Wiki Specs
+
+## Table of contents
+
+1. Motivation
+2. Business rules
+3. Use cases
+4. Data model
+
+## Motivation
+
+>The site contains a variety of users with a vast wealth of knowledge. The wiki is a place where users can put their knowledge to work, and write articles for users to use. These may be supplementary to other documents on the site, or standalone documents such as guides.
+
+## Business rules
+
+> * Articles must be able to be read/write locked based on user class.
+> * Users with the wiki management permission must have full access to all articles.
+> * Article creation must be associated with a permission.
+> * Duplicate aliases, tags, and dedicated terms must be unique.
+
+## Use cases
+
+### 1.0 Creating an article
+
+**Primary Actor:** A user
+
+**Brief:** Users must be able to create wiki articles in order to share knowledge with other users.
+
+**Preconditions:** The user must have the wiki creation permission or the wiki management permission enabled.
+
+**Trigger:** When the user clicks the link to create an article.
+
+**Postconditions:**
+
+* The article is created
+
+**Basic Flow:**
+
+1. The user clicks the link to create an article.
+2. The template is generated, containing a title and section textarea.
+3. The user fills out the fields and user class read/edit restrictions, and hits submit.
+
+**Non-Functional Requirements**
+
+* BBCode must be stored in the database and rendered at page load
+* Each section must have a [+] button to add another section.
+* Each section must have a [++] button to add a sub-section.
+
+### 2.0 Viewing an article
+
+**Primary Actor:** A user
+
+**Brief:** Wiki articles must be able to be viewed.
+
+**Preconditions:** The user has the appropriate user class to read the article, or has the wiki management permission enabled.
```

Comment:
Wiki permissions may need another layer. Really think about how this works. 

Also mention somewhere what needs to happen when a class is deleted which has wiki articles associated with it? We don't have FKs so gotta be extra careful.